### PR TITLE
packaging: drop 99-snapd.conf via dpkg-maintscript-helper

### PR DESCRIPTION
**Pull request by mvo5**
We dropped shipping /etc/sudoers.d/99-snapd.conf in PR#8885 but
did not clean up properly in the debian/* packaging. This is a
conffile in debian packaging terms so it needs special handling.

Fortunately the dpkg-maintscript-helper makes this easy(ish)
now with debian/<pkg>.maintscript

This fixes LP: #1915156

### CODE REVIEW

File: tests/main/upgrade-sudoers-conffile-removal/task.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    The exact version of snapd is a bit arbitrary (anything before
+    2.46 is fine).
+
+systems: [ubuntu-20.04-64]
+
+execute: |
+    #shellcheck source=tests/lib/systemd.sh
+    . "$TESTSLIB"/systemd.sh
+
+    echo "download snapd 2.37.4"
```
Comment (bboozzoo) [inline tests/main/upgrade-sudoers-conffile-removal/task.yaml:16]: s/2.37.4/2.45.1/

File: tests/main/upgrade-sudoers-conffile-removal/task.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    /etc/sudoers.d/99-snapd.conf happens.
+    
+    The exact version of snapd is a bit arbitrary (anything before
+    2.46 is fine).
+
+systems: [ubuntu-20.04-64]
+
+execute: |
+    #shellcheck source=tests/lib/systemd.sh
+    . "$TESTSLIB"/systemd.sh
```
Comment (bboozzoo) [inline tests/main/upgrade-sudoers-conffile-removal/task.yaml:14]: We don't need systemd.sh here, do we?

File: tests/main/upgrade-sudoers-conffile-removal/task.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+summary: Ensure conffile removal from upgrades <2.46 works
```
Comment (bboozzoo) [inline tests/main/upgrade-sudoers-conffile-removal/task.yaml:1]: We have the tests/upgrade to heck upgrades, does it make sense to move this test there?

**Review by bboozzoo — APPROVED** [review]
LGTM with a little test tweaks

**Review by anonymouse64 — APPROVED** [review]
LGTM, thanks for this

**Pull request merged**